### PR TITLE
[Executor][Internal] Refine error message in flow_executor.py

### PR DESCRIPTION
**Pull request by PeiwenGaoMS**
# Description

Refine error message in flow_executor.py

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_core/_errors.py [hunk 1]
```diff
@@ -1,7 +1,13 @@
 from traceback import TracebackException
 
 from promptflow._utils.exception_utils import ADDITIONAL_INFO_USER_EXECUTION_ERROR, last_frame_info
-from promptflow.exceptions import ErrorTarget, UserErrorException, ValidationException
+from promptflow.exceptions import ErrorTarget, SystemErrorException, UserErrorException, ValidationException
+
+
+class NotSupportedError(SystemErrorException):
```
Comment (zzn2) [inline src/promptflow/promptflow/_core/_errors.py:7]: `NotSupported` would be a good error code.

File: src/promptflow/promptflow/executor/_errors.py [hunk 2]
```diff
@@ -137,10 +133,6 @@ class NodeReferenceError(UserErrorException):
     """Exception raised when node reference not found or unsupported"""
 
     def __init__(self, message="", message_format="", target=ErrorTarget.FLOW_EXECUTOR, **kwargs):
-        if message:
-            message = f"Invalid node reference: {message}"
-        elif message_format:
-            message_format = f"Invalid node reference: {message_format}"
```
Comment (zzn2) [inline src/promptflow/promptflow/executor/_errors.py:143]: Do we need to remove `message` and `message_format` from the parameters too?

File: src/promptflow/promptflow/executor/flow_executor.py [hunk 3]
```diff
@@ -646,20 +652,40 @@ def _extract_outputs(self, nodes_outputs, bypassed_nodes, flow_inputs):
                 outputs[name] = flow_inputs[output.reference.value]
                 continue
             if output.reference.value_type != InputValueType.NODE_REFERENCE:
-                raise NotImplementedError(f"Unsupported output type {output.reference.value_type}")
+                raise NotSupportedError(
+                    message_format="The output type '{output_type}' is currently unsupported. "
+                    "Please choose from available type: '{supported_output_type}' and try again.",
```
Comment (zzn2) [inline src/promptflow/promptflow/executor/flow_executor.py:657]: type -> types?

File: src/promptflow/promptflow/_core/_errors.py (hunk 1 above)
Comment (PeiwenGaoMS) [reply src/promptflow/promptflow/_core/_errors.py]: Updated.

File: src/promptflow/promptflow/executor/flow_executor.py (hunk 3 above)
Comment (PeiwenGaoMS) [reply src/promptflow/promptflow/executor/flow_executor.py:657]: Updated.

File: src/promptflow/promptflow/executor/_errors.py (hunk 2 above)
Comment (PeiwenGaoMS) [reply src/promptflow/promptflow/executor/_errors.py:143]: Remove `__init__()` .

**Review by zzn2 — DISMISSED** [review]

**Review by chw-microsoft — APPROVED** [review]

File: src/promptflow/tests/executor/unittests/executor/test_dag_manager.py [hunk 4]
```diff
@@ -75,9 +75,8 @@ def test_pop_bypassed_nodes_with_exception(self):
         with pytest.raises(ReferenceNodeBypassed) as e:
             dag_manager.pop_bypassable_nodes()
         error_message = (
-            "Invalid node reference: The node node2 referenced by node3 has been bypassed, "
-            "so the value of this node cannot be returned. Please refer to the node that will "
-            "not be bypassed as the default return value."
+            "The node 'node2' referenced by 'node3' has been bypassed, so the value of this node cannot be returned. "
```
[the remark this answers is not in the document]
Comment (PeiwenGaoMS) [reply src/promptflow/tests/executor/unittests/executor/test_dag_manager.py]: Updated.

File: src/promptflow/promptflow/executor/flow_executor.py [hunk 5]
```diff
@@ -646,20 +652,40 @@ def _extract_outputs(self, nodes_outputs, bypassed_nodes, flow_inputs):
...
+                    output_type=output.reference.value_type,
+                    supported_output_type=[output_type.value for output_type in InputValueType],
+                )
             node = next((n for n in self._flow.nodes if n.name == output.reference.value), None)
             if not node:
-                raise ValueError(f"Invalid node name {output.reference.value}")
+                raise OutputReferenceNotExist(
```
[the remark this answers is not in the document]
Comment (PeiwenGaoMS) [reply src/promptflow/promptflow/executor/flow_executor.py:665]: Offline synced. Continue to use OutputReferenceNotExist.

**Review by zzn2 — APPROVED** [review]

**Pull request merged**